File: AdityaKumar1010707/BinarySearch/validPerfectSquare.cpp
class Solution {
    public:
        bool isPerfectSquare(int num) {
            if(num ==1) return true;
    
            int start = 1;
            int end = num/2;
    
            bool isPerfectSquare = false;
    
            while(start<=end){
              int mid = (start+end) / 2;
              long long  val = (long long )mid*mid;
              if(val == num) return true;
    
              if(val > num){
                end = mid - 1;
              }
              else{
                start = mid + 1;
              }
            }
    
            return isPerfectSquare;
            
        }
    };

File: AmanRajSingh/BinarySearch/ValidPerfectSquare.java
class Solution {
    public  boolean isPerfectSquare(int num) {
        if (num < 1) return false;
        long start = 1, end  = num;
        
        while (start <= end) {
              long mid = start + (end-start) / 2;
              long square = (long)mid * mid; 

              if(square == num){
                return true;
              }else if(square < num){
                start = mid + 1;
              }
              else {
                end = mid - 1;
              }
        }
      return false;
    }
}